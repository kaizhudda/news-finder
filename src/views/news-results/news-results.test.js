import React from 'react';
import { mount, shallow } from 'enzyme';
import NewsResult from './news-results.ui';

let props = {
  fetchNewsData: jest.fn(),
  hits: [
    {
      created_at: "2018-03-14T03:50:30.000Z",
      title: "Test",
      url: "test",
      author: "Cogito",
      points: 6015,
      story_text: null,
      comment_text: null,
      num_comments: 436,
      story_id: null,
      story_title: null,
      story_url: null,
      parent_id: null,
      created_at_i: 1520999430,
      relevancy_score: 8012
    },
  ],
  filters: {},
  status: ''
};
let wrapped;

describe('NewsResult', () => {
  beforeEach(() => {
    wrapped = mount(<NewsResult {...props} />);
  });

  afterEach(() => {
    wrapped.unmount();
  });

  it('should render the NewsResult Component', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should dispatch fetchNewsData action on load', () => {
    expect(props.fetchNewsData).toHaveBeenCalled();
  });

  describe('when status is loading', () => {

    afterEach(() => {
      props.status = '';
    });
    it('should render Loading Spinner', () => {
      props.status = 'LOADING';
      wrapped = mount(<NewsResult {...props} />);
      expect(wrapped.find('Loader').exists()).toBe(true);
    });
  });

  describe('when filter tag is comment', () => {
    it('should render the story title and comment', () => {
      props.filters.tags = 'comment';
      props.hits[0].story_title = 'Test Comment Title';
      props.hits[0].comment_text = 'Test Comment';
      wrapped = shallow(<NewsResult {...props} />);
      expect(wrapped.find('.title').text()).toEqual(props.hits[0].story_title);
      expect(wrapped.find('ReactMarkdown').exists()).toBe(true);
    });
  });
});