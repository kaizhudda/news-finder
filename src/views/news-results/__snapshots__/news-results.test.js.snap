// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsResult should render the NewsResult Component 1`] = `
<NewsResults
  fetchNewsData={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  filters={Object {}}
  hits={
    Array [
      Object {
        "author": "Cogito",
        "comment_text": null,
        "created_at": "2018-03-14T03:50:30.000Z",
        "created_at_i": 1520999430,
        "num_comments": 436,
        "parent_id": null,
        "points": 6015,
        "relevancy_score": 8012,
        "story_id": null,
        "story_text": null,
        "story_title": null,
        "story_url": null,
        "title": "Test",
        "url": "test",
      },
    ]
  }
  status=""
>
  <div
    className="News-Results row center-lg center-md"
  >
    <div
      className="col-xs-12 col-sm-12 col-md-8 col-lg-7"
    >
      <div
        className="hit-item "
        key="0"
      >
        <span
          className="score"
        >
          6015
        </span>
        <span
          className="title"
        >
          Test
        </span>
        <div
          className="support-hit"
        >
          <span>
            by 
            Cogito
          </span>
          <span>
            a year ago
          </span>
          <span>
            436
             comments
          </span>
        </div>
      </div>
    </div>
  </div>
</NewsResults>
`;
